app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -241,25 +245,38 @@ def build_metric_id(type, resource, metric_id)
     # by ems_ref, which in Hawkular terms is the
     # fully qualified resource path from Hawkular inventory
     def run_generic_operation(operation, ems_ref, parameters = {})
-      with_provider_connection do |connection|
         the_operation = {
           :operationName => operation,
           :resourcePath  => ems_ref.to_s,
           :parameters    => parameters
         }
+        run_operation(the_operation)
```

Comment:
It is because one method executes an operation that actually received an operation as a parameter, but is a different kind of operation (DMR).

I'll rename the variable to dmr_operation and add comments to avoid confusion.